app/controllers/journalist/articles_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Journalist::ArticlesController < ApplicationController
+    before_action :check_journalist, only: [:create, :new]
```

Comment:
I would remove the `only: [:create, :new]`, visitors should not be able to run any of this controllers actions

Reply:
done